src/ray/raylet/worker_pool.cc

```diff
@@ -186,23 +188,46 @@ Process WorkerPool::StartWorkerProcess(const Language &language,
                                    options.end());
         ++dynamic_option_index;
       }
-    } else {
-      size_t num_workers_index = token.find(kWorkerNumWorkersPlaceholder);
-      if (num_workers_index != std::string::npos) {
-        std::string arg = token;
-        worker_command_args.push_back(arg.replace(num_workers_index,
-                                                  strlen(kWorkerNumWorkersPlaceholder),
-                                                  std::to_string(workers_to_start)));
-        num_workers_arg_replaced = true;
-      } else {
-        worker_command_args.push_back(token);
+      continue;
+    }
+
+    if (token == kWorkerRayletConfigPlaceholder) {
+      switch (language) {
+      case Language::JAVA:
+        for (auto &entry : raylet_config_) {
+          std::string arg;
+          arg.append("-Dray.raylet.config.");
+          arg.append(entry.first);
+          arg.append("=");
+          if (entry.first == "num_workers_per_process_java") {
+            arg.append(std::to_string(workers_to_start));
+            num_workers_arg_replaced = true;
+          } else {
+            arg.append(entry.second);
+          }
+          worker_command_args.push_back(arg);
+        }
+        if (!num_workers_arg_replaced) {
+          worker_command_args.push_back(
+              "-Dray.raylet.config.num_workers_per_process_java=" +
+              std::to_string(workers_to_start));
+          num_workers_arg_replaced = true;
+        }
```

Comment:
maybe add a comment here explaining why `num_workers_per_process_java` needs to be handle specially?